src/front/wgsl.rs

```diff
@@ -44,6 +44,28 @@ mod lex {
         input.split_at(pos)
     }
 
+    fn consume_number(input: &str) -> (&str, &str) {
+        let mut is_first_char = true;
```

Comment:
sounds like it should be a state machine, i.e. `enum State` and such...

Reply:
Yeah, I was trying to use an enum state machine, and it just ended up more complex.